stl/inc/algorithm

```diff
@@ -3383,6 +3383,22 @@ _NODISCARD _CONSTEXPR20 _FwdIt1 find_first_of(
     const auto _ULast1  = _STD _Get_unwrapped(_Last1);
     const auto _UFirst2 = _STD _Get_unwrapped(_First2);
     const auto _ULast2  = _STD _Get_unwrapped(_Last2);
+
+    constexpr bool _Is_predicate_equal = _Is_any_of_v<_Pr,
+#if _HAS_CXX20
+        _RANGES equal_to,
+#endif // _HAS_CXX20
+        _STD equal_to<>>;
+
+    if constexpr (_Is_ranges_random_iter_v<remove_const_t<decltype(_UFirst2)>> && _Is_predicate_equal) {
+        const auto _Count2 = _Last2 - _First2;
+        if (_Count2 == 1) {
+            _UFirst1 = _STD _Find_unchecked(_STD move(_UFirst1), _STD move(_ULast1), *_UFirst2);

```

Comment:
This is trying to move from `const auto _ULast1`. We may as well remove constness from `_UFirst2` to simplify away the `remove_const_t` too. (Leaving it on `_ULast2` will be a signal that we haven't forgotten it entirely and are being deliberate.)